src/Exception/ShouldNotHappenException.php

```diff
@@ -0,0 +1,9 @@
+<?php declare(strict_types=1);
+
+namespace Rector\Exception;
+
+use Exception;
+
+final class ShouldNotHappenException extends Exception
```

Comment:
@TomasVotruba Maybe we can find a better name for this Exception?

Reply:
This is from PHPStan inspiration for general "todo" cases that need to be resolved in the future, but are not priority at the moent.

Reply:
Oh, perfect. So, I'll add it to my `TODO` list